### ISSUE TITLE
Avoid having to scroll down again after returning to the previous screen

### ISSUE THREAD

**Issue by LenzGr** [cbeyls/fosdem-companion-android#23]
Your app is truly a life-saver when trying to cope with the massive FOSDEM schedule, thank you for that.

However, there is one thing that somewhat annoys me and causes unnecessary confusion in my opinion: every time I step back one level of detail (e.g. returning to the list of sessions for a given track, or returning to the list of tracks), I end up on top of the list and have to scroll down again to where I was before.

Would it be possible to memorize the location of where the user was in the screen before, and return back to it instead of going to the very top again?

Thank you!

**Comment by cbeyls** [comment]
I'm sorry but I'm unable to reproduce your issue.
There are two ways to leave a session screen: back and up navigation.
The app memorizes the scroll position when you navigate back to the track using the back button.
It launches a new screen and scrolls to the origin session (if necessary) when you use the UP button, as described in the Android navigation guidelines. Because you can press up after tapping a bookmark to see the other sessions  given in the same room.
But in both cases you should always see the session you are navigating from in the list.

Which Android version are you using and which button did you press?

**Comment by cbeyls** [comment]
I think I got it: you are navigating up from the track to the main screen of the app. In that case the main screen is reset. For technical reasons, I cannot change the launch mode of the main screen of an app without screwing up the navigation from the launcher icon, and if I change the behavior of the up button to make it work like a back button, then it will not be possible to reach the main screen of the app after the app has been launched from a notification for example.
In conclusion if you expect to come back at the same place you were before, you should use the back button instead of the up button.

**Comment by LenzGr** [comment]
Thank you for your reply!

FWIW, I was using the App on an LG G3 running Android 5.0.

There were several occasions in which returning to a previous screen caused some confusion on my end.

For example, I frequently used my bookmarks to check for the next upcoming session. Depending on which button I pressed, I either ended up in the track list of that session, or in my bookmark list.

I'd somehow would expect that the "back" button in the app behaves identical to pressing the device's back button. From a UX point of view, I think it's confusing that you land on two different screens, depending on which button you use. I admit I'm not familiar with the Android application design guidelines, but somehow this seems counter-intuitive to me.

If you consider this an "Up" button, not a "back" button, maybe you should consider choosing a different icon?

If I wanted to see the track list when looking at an event's description, how about making it possible to tap the track name displayed in there?

**Comment by cbeyls** [comment]
This is a common misconception from users: that button up there is not the same as the back button - why having 2 buttons doing exactly the same thing ?

I will definitely try to experiment with different launch modes for the main screen so pressing up from the track screen does not re-create it when it already exists in the back stack (and will do exactly the same thing as "back"), but I need to do a lot of tests before I add this because in my experience it can cause unexpected side effects when switching back to the app or re-launching the app from its icon.

Apart from that, I don't really know how I can improve global navigation to make it less confusing to you, all I can tell you is that I respect Android navigation guidelines to the letter.

For more information about back versus up navigation:
- [Material design guidelines](https://material.io/guidelines/patterns/navigation.html#navigation-up-back-buttons)
- [Navigation with Back and Up](https://developer.android.com/design/patterns/navigation.html)
